File: poc-kronos-frontend/src/model/matricula/Matricula.js
class Matricula {
  static PRE_MATRICULA() {
    return 1;
  }
  static CANCELADA() {
    return 6;
  }
  static TRANSFERIDA() {
    return 5;
  }
  static VALIDADA() {
    return 2;
  }
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataInicioCurso = dados ? dados.dataInicioCurso : null;
    this.dataConclusaoCurso = dados ? dados.dataConclusaoCurso : null;
    this.data = dados ? dados.data : null;
    this.anoInicioCurso = dados ? dados.anoInicioCurso : null;
    this.semestreInicioCurso = dados ? dados.semestreInicioCurso : null;
    this.dataColacaoGrau = dados ? dados.dataColacaoGrau : null;
    this.anoConclusaoCurso = dados ? dados.anoConclusaoCurso : null;
    this.semestreConclusaoCurso = dados ? dados.semestreConclusaoCurso : null;
    this.idPessoa = dados ? dados.idPessoa : null;
    this.numeroRegistroPessoa = dados ? dados.numeroRegistroPessoa : null;
    this.cpfPessoa = dados ? dados.cpfPessoa : null;
    this.nomePessoa = dados ? dados.nomePessoa : null;
    this.dataNascimentoPessoa = dados ? dados.dataNascimentoPessoa : null;
    this.idGeneroPessoa = dados ? dados.idGeneroPessoa : null;
    this.idArqAnexoPessoa = dados ? dados.idArqAnexoPessoa : null;
    this.idCurso = dados ? dados.idCurso : null;
    this.nomeCurso = dados ? dados.nomeCurso : null;
    this.idTipoSituacaoMatricula = dados ? dados.idTipoSituacaoMatricula : null;
    this.nomeTipoSituacaoMatricula = dados
      ? dados.nomeTipoSituacaoMatricula
      : null;
    this.idTipoResultado = dados ? dados.idTipoResultado : null;
    this.nomeTipoResultado = dados ? dados.nomeTipoResultado : null;
    this.idEmpresaOrigem = dados ? dados.idEmpresaOrigem : null;
    this.nomeInstituicaoOrigem = dados ? dados.nomeInstituicaoOrigem : null;

  }

  static validations(required, numeric, dataFimMenor) {
    return {
      dataInicioCurso: { required },
      dataConclusaoCurso: { dataFimMenor: dataFimMenor("dataInicioCurso") },
      data: { required },
      anoInicioCurso: { required, numeric },
      semestreInicioCurso: { required, numeric },
      anoConclusaoCurso: { numeric },
      semestreConclusaoCurso: { numeric },
      idPessoa: { required },
      idCurso: { required }
      // idEmpresaOrigem: { required }
    };
  }

  static validationMessages() {
    return {
      dataInicioCurso: {
        required: "Data Início do Curso é obrigatório"
      },
      dataConclusaoCurso: {
        dataFimMenor:
          "Data Conclusão do Curso não pode ser menor que a Data Conclusão do Curso"
      },
      data: {
        required: "Data é obrigatório"
      },
      anoInicioCurso: {
        required: "Ano início é obrigatório",
        numeric: "Somente números"
      },
      semestreInicioCurso: {
        required: "Semestre início é obrigatório",
        numeric: "Somente números"
      },
      anoConclusaoCurso: {
        numeric: "Somente números"
      },
      semestreConclusaoCurso: {
        numeric: "Somente números"
      },
      idPessoa: {
        required: "Pessoa é obrigatória"
      },
      idCurso: {
        required: "Curso é obrigatório"
      }
      // idEmpresaOrigem: {
      //   required: "Empresa origem é obrigatória"
      // }
    };
  }
}
export default Matricula;

File: src/main/java/br/com/kronos/backend/aplicacao/matricula/Matricula.java
package br.com.kronos.backend.aplicacao.matricula;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Matricula {
	
	public static final String COMBO_CACHE_NAME_OCORRENCIAS_POR_PESSOA = "combo_matriculas_ocorrencias_por_pessoa";
    
    private long id;
    private LocalDate data;
    private Integer anoInicioCurso;
    private Integer semestreInicioCurso;
    private LocalDate dataInicioCurso;
    private Integer anoConclusaoCurso;
    private Integer semestreConclusaoCurso;
    private LocalDate dataConclusaoCurso;
    private LocalDate dataColacaoGrau;
    private Long idPessoa;
    private Long idCurso;
    private Integer idTipoSituacaoMatricula;
    private Integer idTipoResultado;
    private Long idEmpresaOrigem;

	@Builder
	public Matricula(long id, LocalDate data, Integer anoInicioCurso, Integer semestreInicioCurso, LocalDate dataInicioCurso, Integer anoConclusaoCurso, 
			Integer semestreConclusaoCurso, LocalDate dataConclusaoCurso, LocalDate dataColacaoGrau,  Long idPessoa, Long idCurso, 
                     Integer idTipoSituacaoMatricula, Integer idTipoResultado, Long idEmpresaOrigem) {

        this.id = id;
        this.data = exigirNaoNulo(data, "Data");
        this.anoInicioCurso = exigirNaoNulo(anoInicioCurso, "Ano de início do curso");
        this.semestreInicioCurso = exigirNaoNulo(semestreInicioCurso, "Semestre de início do curso");  
        this.dataInicioCurso = exigirNaoNulo(dataInicioCurso, "Data de início do curso"); 
        this.anoConclusaoCurso = anoConclusaoCurso; 
        this.semestreConclusaoCurso = semestreConclusaoCurso;  
        this.dataConclusaoCurso = dataConclusaoCurso; 
        this.dataColacaoGrau = dataColacaoGrau;  
        this.idPessoa = exigirNaoNulo(idPessoa, "Aluno");
        this.idCurso = exigirNaoNulo(idCurso, "Curso");  
        this.idTipoSituacaoMatricula = exigirNaoNulo(idTipoSituacaoMatricula, "Situação da matrícula");  
        this.idTipoResultado = idTipoResultado; 
        this.idEmpresaOrigem = idEmpresaOrigem;  

	}
}
